src/core/util/threadpool.cpp

```diff
@@ -43,18 +43,26 @@ ThreadPool::ThreadPool(size_t threads, std::function<void()> onThreadStart,
     }
 }
 
+ThreadPool::~ThreadPool() {
+    std::lock_guard<std::mutex> lock(workers_mutex);
```

Comment:
This might also deadlock if any of the currently running jobs want to dispatch something before finishing I think